src/main/java/com/example/bookshop/service/impl/OrderItemServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package com.example.bookshop.service.impl;
+
+import com.example.bookshop.dto.orderitem.OrderItemDto;
+import com.example.bookshop.exception.EntityNotFoundException;
+import com.example.bookshop.mapper.OrderItemMapper;
+import com.example.bookshop.model.Order;
+import com.example.bookshop.model.OrderItem;
+import com.example.bookshop.repository.cartitem.CartItemRepository;
+import com.example.bookshop.repository.orderitem.OrderItemRepository;
+import com.example.bookshop.service.OrderItemService;
+import com.example.bookshop.service.ShoppingCartService;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderItemServiceImpl implements OrderItemService {
+    private final ShoppingCartService shoppingCartService;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderItemRepository orderItemRepository;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    public Set<OrderItem> getOrderItems() {
+        return cartItemRepository.findAllByShoppingCartId(shoppingCartService
+                .getShoppingCart().id()).stream()
+                .map(c -> {
+                    OrderItem orderItem = orderItemMapper.convertCartItemToOrderItem(c);
+                    orderItem.setPrice(orderItem.getBook().getPrice()
+                            .multiply(new BigDecimal(orderItem.getQuantity())));
+                    return orderItem;
+                }).collect(Collectors.toSet());

```

Comment:
it's hard to read. can you create some separate variables or move some logic to a private method?